eng/PrepareRelease.yml

```diff
@@ -0,0 +1,53 @@
+stages:
+- stage: PrepareReleaseStage
+  variables:
+    isRelease: $[in(variables['Build.SourceBranch'], 'refs/heads/release/')]
+  displayName: Release Preparation
+  dependsOn:
+  - publish_using_darc
+  jobs:
+  - job: PrepareReleaseJob
+    displayName: Prepare release with Darc
+    pool: 
+      vmImage: windows-latest
+    variables:
+    - group: DotNet-Diagnostics-Storage

```

Comment:
These are only meant to be mapped in internal please. Can we split this file in two sections too please?:
```yml
# bunch of metadata and stage and job defs.
steps:
- ${{ if in(variables['Build.Reason'], 'PullRequest') }}:
  - # simple build case.
- ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest'), not(in(variables['Build.Reason'], 'PullRequest')) }}:
  -  real publish
```